### DISCUSSION TITLE
Support for new image formats like AVIF, JPEGXL, HEIC

### DISCUSSION THREAD

**Discussion by shubhendusaurabh** [dlemstra/magick-wasm#36] [Ideas]
Would love to have encoding/decoding capability for new image formats. magick-wasm supports heic reading but not writing.

- AVIF
- JPEGXL
- HEIC

Would love to have support for them in magick-wasm.

**Comment by dlemstra** [comment]
Support for writing `HEIC` files is not possible due to licensing issues. But it might be possible to add support for `JPEGXL` and `AVIF`.

**Reply by shubhendusaurabh** [reply]
Are their support on the roadmap?

Thanks.

**Reply by dlemstra** [reply]
I just check if it was possible to add support for aom (required for `avif`) and that turned out to be easy. Adding support for `jpeg-xl` will probably take more time.

**Comment by dlemstra** [comment]
A new release that supports `avif` was just published.

**Reply by dlemstra** [reply]
And I just tried to add support for `jpeg-xl` and that turned out to be easy now (it wasn't last time I tried). This will become available in the next release.

**Reply by shubhendusaurabh** [reply]
i tried loading jxl files but getting runtime aborted error.
![index](https://user-images.githubusercontent.com/1818387/207442148-19af4d6c-f66f-46d3-a265-cf6f3093ffa4.png)

**Reply by dlemstra** [reply]
Can you share an image that can be used to reproduce this?

**Reply by shubhendusaurabh** [reply]
I had converted some images using https://squoosh.app/ using jpeg xl encoder. Also tried reading some images from this page: https://people.csail.mit.edu/ericchan/hdr/hdr-jxl.php

another issue i found: I can read avif files but getting blank output when converting images to avif.

**Reply by dlemstra** [reply]
This might be related to recent upgrades of dependencies. Will need to investigate this when I have some time.

**Reply by dlemstra** [reply]
Found what is causing the issue with the jpeg-xl images. I pushed a patch to the ImageMagick repository to resolve this issue and this will be fixed in the next release. And can you open a separate issue for the avif write problem with a small code sample?
